Bug: load-model.sh does not print separator or error message when a script fails

## Bug

`load-model.sh` exits via `set -e` before printing the separator or error message when a script fails.

## Failing test

- **Test file:** `pipeline/tests/test_integration.py`
- **Test name:** `TestLoadModelSh::test_script01_fails_separator_present`
- **Test name:** `TestLoadModelSh::test_script01_fails_error_message_present`
- **Test name:** `TestLoadModelSh::test_script02_fails_error_message_present`

## Full error output

```
AssertionError: assert '--- last output from 01_download_corpus.py ---' in ('Running 01_download_corpus.py...\n' + '')
 +  where 'Running 01_download_corpus.py...\n' = CompletedProcess(..., returncode=1, stdout='Running 01_download_corpus.py...\n', stderr='').stdout
 +  and   '' = CompletedProcess(..., returncode=1, ..., stderr='').stderr

AssertionError: assert 'ERROR: 01_download_corpus.py failed with exit code 1' in ('Running 01_download_corpus.py...\n' + '')
```

## Expected vs actual

- **Expected:** When a Python script exits non-zero, `load-model.sh` prints `--- last output from <name> ---`, replays the last 10 lines, then prints `ERROR: <name> failed with exit code N`, then exits with code N.
- **Actual:** The script exits with the correct non-zero code but prints only `Running <name>...` — the separator and error lines are never printed.

## Likely implementation file

`pipeline/load-model.sh`

## Root cause

`load-model.sh` uses `set -euo pipefail`. The `run_script` function runs:

```bash
python3 "$name" 2>&1 | tee "$tmpfile"
local code=${PIPESTATUS[0]}
```

With `pipefail`, when `python3` exits non-zero, the pipeline exits non-zero. With `set -e`, bash exits the function immediately at the pipeline line — `local code=${PIPESTATUS[0]}` and the `if` block that prints the separator and error message are never reached.

## Relevant acceptance criterion

From spec `docs/specs/feature-112/load-model-sh.md`:

> Script 01 fails with exit code 1: exit code 1; stdout contains `"--- last output from 01_download_corpus.py ---"` followed by `"ERROR: 01_download_corpus.py failed with exit code 1"`; script 02 not invoked

--- a/pipeline/load-model.sh
+++ b/pipeline/load-model.sh
@@ -6,8 +6,10 @@ run_script() {
   local tmpfile
   tmpfile=$(mktemp)
   echo "Running ${name}..."
+  set +e
   python3 "$name" 2>&1 | tee "$tmpfile"
   local code=${PIPESTATUS[0]}
+  set -e
   if [ "$code" -ne 0 ]; then
     echo "--- last output from ${name} ---"
     tail -10 "$tmpfile" || true

